File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program136.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and count number of ON (1) bits in it without using % and / operator.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef unsigned int UINT;

class Solution
{
	public:
		UINT CountOne(UINT iNo)
		{
			UINT iMask = 0x00000001;
			UINT iResult=0,i=0,iCnt=0;

			for(i=1;i<32;i++)
			{
				iResult=iNo&iMask;
				if(iResult!=0)
				{
					iCnt++;
				}
				iMask=iMask<<1;
			}

			return iCnt;
		}
};

int main()
{

	UINT iValue=0,uRet=0;

	cout<<"Enter number: ";
	cin>>iValue;

	Solution *sObj = new Solution();
	uRet = sObj->CountOne(iValue);

	cout<<"Number of ON bits are: "<<uRet<<"\n";
	return 0;
}

/*
OUTPUT:

Enter number: 11
Number of ON bits are: 3

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program136.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and count number of ON (1) bits in it without using % and / operator.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Solution
{

	int CountOne(int iNo)
	{
		int iMask = 0x00000001,iResult=0,iCnt=0;

		for(int i=1;i<32;i++)
		{
			iResult = iNo&iMask;

			if(iResult!=0)
			{
				iCnt++;
			}

			iMask = iMask<<1;
		}

		return iCnt;

	}	
}

class Program
{
	public static void main(String[] args)
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 	
		
		Solution sObj1 = new Solution();
		iRet = sObj1.CountOne(iValue);

		System.out.println("Number of ON bits are: "+iRet);
	}	
}
/*
OUTPUT:

Enter number: 
11
Number of ON bits are: 3

*/
















